### PR TITLE
Upgrade to the latest viper release

### PR DESCRIPTION
**Pull request by bogdandrutu**
Changes in this PR:
* Rewrite Viper.Sub because of an issue documented in the comment;
* Use config.NewViper everywhere to enforce a different key-delimiter "::" that allows dots (".") in the config keys.
* Remove unnecessary hack to copy the entire Viper config when doing environment variables expanding.

### CODE REVIEW

File: receiver/jaegerreceiver/factory_test.go [hunk 1]
```diff
@@ -275,15 +275,15 @@ func TestRemoteSamplingFileRequiresGRPC(t *testing.T) {
 
 func TestCustomUnmarshalErrors(t *testing.T) {
 	factory := Factory{}
-	v := viper.New()
```
Comment (pjanotti) [inline receiver/jaegerreceiver/factory_test.go]: when this change gets to contribs we will have to check for these usages too.

File: config/config.go [hunk 2]
```diff
@@ -862,28 +870,18 @@ func validateProcessors(cfg *configmodels.Config) {
...
+	sv := viperSub(v, key)
+	expandEnvConfig(sv)
 	// Before unmarshaling first expand all environment variables.
-	return expandEnvConfig(sv)
+	return sv
 }
 
 // expandEnvConfig creates a new viper config with expanded values for all the values (simple, list or map value).
 // It does not expand the keys.
-// Need to copy everything because of a bug in Viper: Set a value "map[string]interface{}" where a key has a ".",
```
Comment (pjanotti) [inline config/config.go]: I assume there is already a test covering this case and the change is already covered by that.

File: receiver/jaegerreceiver/factory_test.go (hunk 1 above)
Comment (bogdandrutu) [reply receiver/jaegerreceiver/factory_test.go]: +1

File: config/config.go (hunk 2 above)
Comment (bogdandrutu) [reply config/config.go]: Yes.

**Pull request merged**